File: Backend/Mono/Nonagon.Modular.DynamicForm/FormLayoutWidth.cs
using System;

namespace Nonagon.Modular.DynamicForm
{
	/// <summary>
	/// Form layout size in horizontal.
	/// </summary>
	public enum FormLayoutWidth
	{
		/// <summary>
		/// The size is growing automatically by the children.
		/// </summary>
		WrapContent,
		
		/// <summary>
		/// The size adjusted to be the parent size (exclude parent padding and it's margin).
		/// </summary>
		MatchParent,
		
		/// <summary>
		/// The size adjusted to be average with their sibling children.
		/// </summary>
		Average
	}
}

File: Frontend/GWT/gwt-modular-dynamicform/src/com/nonagon/modular/dynamicform/client/FormLayoutWidth.java
package com.nonagon.modular.dynamicform.client;

public enum FormLayoutWidth {

	WrapContent,
	MatchParent,
	Average;
	
	public static FormLayoutWidth fromString(String value) {

		if(value == null)
			return MatchParent;
			
		return Enum.valueOf(FormLayoutWidth.class, value);
	}
	
	public static String fromEnum(FormLayoutWidth formLayoutWidth) {
		
		return formLayoutWidth.toString();
	}
}
